test/uk/gov/hmrc/hec/controllers/SDESCallbackControllerSpec.scala

```diff
@@ -0,0 +1,227 @@
+
+        "the JSON in the call back cannot be parsed" in {
+          status(performActionWithJsonBody(JsString("hi"))) shouldBe BAD_REQUEST
+        }
+      }
+
+      "return a 500( internal server error)" when {
+
+        "file deletion from object store fails" in {
+          mockDeleteFile(fileName1, "sdes/licence-type")(Left(Error("err")))
+          val notification =
+            createCallBackNotification(FileProcessingFailure, fileName1)
+          val result       = performActionWithJsonBody(Json.toJson(notification))
+          status(result) shouldBe INTERNAL_SERVER_ERROR
+        }
+
+        "call to fetch records from mongo db fails" in {
+          mockDeleteFile(fileName2, "sdes/tax-checks")(Right(()))
+          mockGetAllTaxCheckCodesByCorrelationId(uuid)(Left(Error("err")))
+          val notification =
+            createCallBackNotification(FileProcessed, fileName2)
+          val result       = performActionWithJsonBody(Json.toJson(notification))
+          status(result) shouldBe INTERNAL_SERVER_ERROR
+        }
+
+        "call to update hec tax checks records fails" in {
+          mockDeleteFile(fileName2, "sdes/tax-checks")(Right(()))
+          mockGetAllTaxCheckCodesByCorrelationId(uuid)(Right(List(hecTaxCheck1, hecTaxCheck2)))
+          mockUpdateAllHecTaxCheck(
+            List(
+              hecTaxCheck1.copy(fileCorrelationId = None, isExtracted = true),
+              hecTaxCheck2.copy(fileCorrelationId = None, isExtracted = true)
+            )
+          )(Left(Error("err")))
+          val notification =
+            createCallBackNotification(FileProcessed, fileName2)
+          val result       = performActionWithJsonBody(Json.toJson(notification))
+          status(result) shouldBe INTERNAL_SERVER_ERROR
+        }
+
+      }
+
+      "return 200" when {
+
+        "call back status is FileReady, no data processing" in {
+          val notification =
+            createCallBackNotification(FileReady, fileName1)
+          val result       = performActionWithJsonBody(Json.toJson(notification))
+          status(result) shouldBe OK
+        }
+
+        "call back status is FileReceived, no data processing" in {
+          val notification =
+            createCallBackNotification(FileReceived, fileName1)
+          val result       = performActionWithJsonBody(Json.toJson(notification))
+          status(result) shouldBe OK
+        }
+
+        "call back status is FileProcessingFailure, delete the file from object store" in {
+          mockDeleteFile(fileName1, "sdes/licence-type")(Right(()))
+          val notification =
+            createCallBackNotification(FileProcessingFailure, fileName1)
+          val result       = performActionWithJsonBody(Json.toJson(notification))
+          status(result) shouldBe OK
+        }
+
+        "call back status is FileProcessed and " when {
+
+          "and the file is not the hec tax check file, only delete the file" in {
```

Comment:
should we delete the file if it's not a hec tax check file? shouldn't we just ignore it?

Reply:
we have to delete all processed files from the object store, if it's not hec tax check file, then it must be an  ENUm files. We have to delete those as well

Reply:
the deletion here is mostly just to clean up after ourselves and free up memory for everyone else. In reality the enum files are going to be very small, but it seems like good practice to clean up things you'll never use again in the future

Reply:
i can do that

Reply:
[updated](https://github.com/hmrc/hec/pull/42/commits/d8d8084cb11f34742f3758ef31cb5a706ab4bb20)